### PR TITLE
Graduate AdmissionWebhookMatchConditions to beta

### PR DESCRIPTION
**Pull request by a-hilaly**
This PR updates the kubernetes documentation to state that `AdmissionWebhookMatchConditions` is currently in Beta stage.

Graduation PR: https://github.com/kubernetes/kubernetes/pull/119380

### CODE REVIEW

File: content/zh-cn/docs/reference/command-line-tools-reference/feature-gates.md
Comment (dipesh-rawat) [inline content/zh-cn/docs/reference/command-line-tools-reference/feature-gates.md:1]: @A-Hilaly It appears that this PR includes changes to files across multiple languages. However, docs follows different processes for each localization, and and we typically don't accept pull requests that impact multiple languages simultaneously. I would recommend modify only the primary website content (which means only English content) initially with this PR. After the merge of the English content, we could create separate pull request for the remaining languages. I hope this suggestion is helpful.

File: api-ref-assets/api/swagger.json
Comment (sftim) [inline api-ref-assets/api/swagger.json:1]: Don't change this file other than during our documented release process.

**Comment by a-hilaly** [comment]
@katcosgrove @sftim @dipesh-rawat @ivelichkovich  I just addressed all the comments and changed target branch to `dev-1.28`.

File: content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md:1]: For graduation to beta, we'd like to see the docs looking more polished than alpha.

Ideally: convert the example ValidatingWebhookConfiguration into a manifest using a `codenew` shortcode.

See https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#source-code-files for details, or have a look at how other documentation does it.

**Review by sftim — COMMENTED** [review]
Thanks

File: content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md
Comment (a-hilaly) [reply content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md:1]: Done! @sftim

**Comment by a-hilaly** [comment]
/remove-language de fr es hi id ja pl pt ru vi

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -68,7 +68,8 @@ For a reference to old feature gates that are removed, please refer to
 | `APIServerIdentity` | `true` | Beta | 1.26 | |
 | `APIServerTracing` | `false` | Alpha | 1.22 | 1.26 |
 | `APIServerTracing` | `true` | Beta | 1.27 | |
-| `AdmissionWebhookMatchConditions` | `false` | Alpha | 1.27 | |
+| `AdmissionWebhookMatchConditions` | `false` | Alpha | 1.27 | 1.27 |
```
Comment (cici37) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:71]: Extra column of `1.27`?

File: content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md [hunk 2]
```diff
@@ -721,7 +721,7 @@ The `matchPolicy` for an admission webhooks defaults to `Equivalent`.
 
 ### Matching requests: `matchConditions`
 
-{{< feature-state state="alpha" for_k8s_version="v1.27" >}}
+{{< feature-state state="beta" for_k8s_version="v1.28" >}}
```
Comment (cici37) [inline content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md:724]: Do we want to add information on the limit we set(65) on MatchConditions into the doc?

**Review by cici37 — COMMENTED** [review]
The tech part generally lgtm. Just couple nits.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 1 above)
Comment (a-hilaly) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:71]: The last element is for the `until` column - just stating that the feature was alpha in 1.27

File: content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md (hunk 2 above)
Comment (a-hilaly) [reply content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md:724]: Added a note stating the 64 elements limit

File: content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md:1]: Is this still true:
> Use of `matchConditions` requires the [featuregate](/docs/reference/command-line-tools-reference/feature-gates/) `AdmissionWebhookMatchConditions` to be explicitly enabled on the kube-apiserver before this feature can be used.

**Comment by sftim** [comment]
BTW, the assigned reviewers are actually @asa3311 and @kbhawkey; I'm just helping out.

**Comment by krol3** [comment]
can you confirm is related KEP: https://github.com/kubernetes/enhancements/issues/3716 ? @a-hilaly

**Comment by a-hilaly** [comment]
>can you confirm is related KEP: https://github.com/kubernetes/enhancements/issues/3716 ? @a-hilaly

@krol3 yes, i confirm.

**Comment by mickeyboxell** [comment]
@a-hilaly Could you take a look at the feedback above at your earliest convenience?

File: content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md [hunk 3]
```diff
@@ -736,55 +731,11 @@ webhook to be called.
...
-    matchConditions:
-      - name: 'breakglass'
-        # Skip requests made by users authorized to 'breakglass' on this webhook.
-        # The 'breakglass' API verb does not need to exist outside this check.
-        expression: '!authorizer.group("admissionregistration.k8s.io").resource("validatingwebhookconfigurations").name("my-webhook.example.com").check("breakglass").allowed()'
-```
+{{< codenew file="access/validating-webhook-configuration-match-conditions.yaml" >}}
+
+{{< note >}}
+You can define up to 64 elements in the `matchConditions`` field per webhook.
```
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md:737]:
```suggestion
You can define up to 64 elements in the `matchConditions` field per webhook.
```

**Review by sftim — COMMENTED** [review]
One suggestion, one syntax error

**Comment by a-hilaly** [comment]
@sftim addressed your comment!

File: content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md [hunk 4]
```diff
@@ -721,12 +721,7 @@ The `matchPolicy` for an admission webhooks defaults to `Equivalent`.
...
 
-{{< feature-state state="alpha" for_k8s_version="v1.27" >}}
-
-{{< note >}}
-Use of `matchConditions` requires the [featuregate](/docs/reference/command-line-tools-reference/feature-gates/)
-`AdmissionWebhookMatchConditions` to be explicitly enabled on the kube-apiserver before this feature can be used.
-{{< /note >}}
+{{< feature-state state="beta" for_k8s_version="v1.28" >}}
 
 You can define _match conditions_for webhooks if you need fine-grained request filtering. These
```
Comment (Rishit-dagli) [inline content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md:726]: Not changed by this PR but this in the document causes it be rendered incorrectly

```suggestion
You can define _match conditions_ for webhooks if you need fine-grained request filtering. These
```

**Review by Rishit-dagli — CHANGES_REQUESTED** [review]

**Comment by a-hilaly** [comment]
@Rishit-dagli  Page is rendering well at https://deploy-preview-42060--kubernetes-io-vnext-staging.netlify.app/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-matchconditions

**Review by Rishit-dagli — COMMENTED** [review]
/lgtm

**Pull request merged**